### PR TITLE
Reconstruction test

### PR DESCRIPTION
**Pull request by soroushzargar**
This is a more modular and well structured re-written version of the reconstruction test. There are also some more modifications I should do which I will in a few days. I should reduce its dependencies and some other modifications. But the current code is a way better than what I wrote previous time.

### CODE REVIEW

File: test/test_reconstruction.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/python
+"""
+Details about the Doument: ///// COMPLETE
```
Comment (danpovey) [inline test/test_reconstruction.py]: is this or reconstructin-test supposed to be here?    Seems to be a duplication.

**Comment by soroushzargar** [comment]
Yes in the branch I totally removed the previous file and wrote the code
again.

**Comment by soroushzargar** [comment]
But I don't know why in thw merge it seemed to be redundant, maybe there is
a problem with removal. It may not have been removed in commits.

File: test/test_reconstruction.py (hunk 1 above)
Comment (soroushzargar) [reply test/test_reconstruction.py]: Yes in the branch I totally removed the previous file and wrote the code
again.

**Comment by danpovey** [comment]
I wanted to try removing whichever one was not supposed to be there but I can't find out which is the real one because there is no run.sh that would indicate the correct usage to test it.
It's just `git rm` to remove a file.

**Comment by soroushzargar** [comment]
I don't know what was the cause of this but I handled it. Now it is clear. The file is runnable. Actually, I did not rename it to `run.sh` in order to be in the same shape as other tests.

**Comment by danpovey** [comment]
If it is a python script it should have the suffix .py.

Perhaps I was not clear.  I want there to be a top-level shell script- say, run.sh, or run_comparison.sh  - that downloads the data from openslr if it is not already downloaded, and runs that python script with appropriate args.  Right now it's not at all easy for someone unfamiliar with the project to do that.
So the user would just do
```
cd tests
 ./run_comparison.sh
```
to run that test.
Also please finish the doc-string at the top, right now it says COMPLETE.
Let me know if you think you understand what I am asking you to do.

**Comment by soroushzargar** [comment]
Yes, Actually. So by now, just the script should download the dataset (if
not downloaded) and place it near the code, then it should run the test (as
it does).
At last the docstring should be completed. Ok, I'm on it. Just there might
be a problem for me today, In case I will try to deliver it tomorrow.

Soroush Haj Zargarbashi
M.Sc in Computer Science
University of Tehran

**Comment by soroushzargar** [comment]
some errors in different sample rates occurred. BTW the shell file is being written. It also installs all the needed packages in case of absence.

**Comment by soroushzargar** [comment]
I'll go for checking the problem with sample rate I'll create another PR if you considered merging this.

File: test/run_comparison.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+ #!/bin/bash
+echo Checking the packages to be installed
+
+pip3 show numpy
```
Comment (danpovey) [inline test/run_comparison.sh:4]: For this, I think you should be able to use a requirements.txt and do it all in one go.

Comment (danpovey) [reply test/run_comparison.sh:4]: I didn't realize the python script did the downloading etc., itself.  Perhaps just a README explaining that you can just run it directly would have been enough.  I'll merge, we can worry about it later anyway.

**Comment by danpovey** [comment]
I pushed a change RE readability.
I think you may be just trusting that mp3 is using the bitrates that  you
direct?  It may not, at all sample rates; you should measure the num-bytes
in the file.

**Comment by soroushzargar** [comment]
Sorry, but I thought that the bitrate name as standard. If you wouldn't mind I'll fix this in a day or two. Sorry but now it's a super hard time for me. I'll get to it soon.

**Comment by danpovey** [comment]
Thanks.

**Pull request merged**